app/products/agents/store/streaming_store.ts

```diff
@@ -0,0 +1,265 @@
+            reasoning: '',
+            isReasoningLoading: false,
+            showReasoning: false,
+            toolCalls: [],
+            annotations: [],
+        };
+
+        this.streamingPosts.set(postId, state);
+        this.emitEvent(StreamingEvents.STARTED, state);
+    }
+
+    /**
+     * Update the streaming message
+     */
+    updateMessage(postId: string, message: string): void {
+        const state = this.streamingPosts.get(postId);
+        if (!state) {
+            return;
+        }
+
+        state.message = message;
+        state.precontent = false;
+        state.generating = true;
+
+        this.emitEvent(StreamingEvents.UPDATED, state);
+    }
+
+    /**
+     * End streaming for a post
+     */
+    endStreaming(postId: string): void {
+        const state = this.streamingPosts.get(postId);
+        if (!state) {
+            return;
+        }
+
+        state.generating = false;
+        state.precontent = false;
+        state.isReasoningLoading = false; // Stop showing reasoning loading state
+
+        this.emitEvent(StreamingEvents.ENDED, state);
+
+        // Clean up immediately - component clears its local state on ENDED event
+        this.streamingPosts.delete(postId);
+    }
+
+    /**
+     * Update reasoning summary
+     */
+    updateReasoning(postId: string, reasoning: string, isLoading: boolean): void {
+        const state = this.streamingPosts.get(postId);
+        if (!state) {
+            return;
+        }
+
+        state.reasoning = reasoning;
+        state.isReasoningLoading = isLoading;
+        state.showReasoning = true;
+
+        // During reasoning, explicitly set generating to false to prevent blinking cursor
+        if (isLoading) {
+            state.generating = false;
+            state.precontent = false;
+        }
+
+        this.emitEvent(StreamingEvents.UPDATED, state);
+    }
+
+    /**
+     * Update tool calls
+     */
+    updateToolCalls(postId: string, toolCallsJson: string): void {
+        let state = this.streamingPosts.get(postId);
+
+        // If no streaming state exists, create a minimal one for tool call updates
+        // This handles tool status updates that arrive after streaming ends but before POST_EDITED
+        if (!state) {
+            state = {
+                postId,
+                generating: false,
+                message: '',
+                precontent: false,
+                reasoning: '',
+                isReasoningLoading: false,
+                showReasoning: false,
+                toolCalls: [],
+                annotations: [],
+            };
+            this.streamingPosts.set(postId, state);
+        }
+
+        try {
+            state.toolCalls = JSON.parse(toolCallsJson);
+            state.precontent = false;
+            this.emitEvent(StreamingEvents.UPDATED, state);
+        } catch (error) {
+            logWarning('Failed to parse tool calls JSON', error, {postId, toolCallsJson});
```

Comment:
For the logs, we normally tend to add a prefix with the function name to know where is it from